### PR TITLE
Document the `prune_code_paths` option for the compiler

### PR DESCRIPTION
**Pull request by saleyn**

### CODE REVIEW

File: lib/mix/lib/mix/tasks/compile.ex [hunk 1]
```diff
@@ -35,6 +35,15 @@ defmodule Mix.Tasks.Compile do
...
 
+    * `:prune_code_paths` - prune code paths before compilation. When true
+      (default), this prunes code paths of applications that are not listed
+      in the project file with dependencies.  When false, this keeps the
+      entirety of Erlang/OTP available on the project starts, including
+      the paths set by the code loader from the `ERL_LIBS` environment as
+      well as explicitely listed by providing `-pa` and `-pz` options
+      to Erlang.
+
+
```
Comment (josevalim) [inline lib/mix/lib/mix/tasks/compile.ex:46]:
```suggestion

```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

File: lib/mix/lib/mix/tasks/compile.ex [hunk 2]
```diff
@@ -35,6 +35,13 @@ defmodule Mix.Tasks.Compile do
       this has undesirable side-effects (such as skipping some
       compiler checks) and should be avoided.
 
+    * `:prune_code_paths` - prune code paths before compilation. When true
+      (default), this prunes code paths of applications that are not listed
+      in the project file with dependencies.  When false, this keeps the
+      entirety of Erlang/OTP available on the project starts, including
```
Comment (dimitarvp) [inline lib/mix/lib/mix/tasks/compile.ex:41]: Should be "on project start" here, or "when the project starts".

Comment (saleyn) [reply lib/mix/lib/mix/tasks/compile.ex:41]: I literally copied the existing phrase from a few lines below. ;)  Fixing in both places...

Comment (dimitarvp) [reply lib/mix/lib/mix/tasks/compile.ex:41]: Eh, too late now, just felt the need to point at the correct English phrasing. Next time!

Comment (saleyn) [reply lib/mix/lib/mix/tasks/compile.ex:41]: Submitted a separate PR ;)

**Pull request merged**